app/src/main/java/com/sun/ise/data/local/LocalDataSource.kt

```diff
@@ -0,0 +1,18 @@
+package com.sun.ise.data.local
+
+import android.content.Context
+import com.sun.ise.data.UserDataSource
+import com.sun.ise.util.SharePrefs
+
+class LocalDataSource(context: Context) : UserDataSource.Local {
```

Comment:
nên để localDataSource là singleton để đảm bảo về vấn đề thread-safe suggest và dễ mantain hơn
https://github.com/googlesamples/android-architecture/blob/dev-todo-mvvm-rxjava/todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/data/source/local/TasksLocalDataSource.java
```suggestion
  companion object {
        private var INSTANCE: TasksLocalDataSource? = null

        @JvmStatic
        fun getInstance(context: Context): TasksLocalDataSource {
            if (INSTANCE == null) {
                synchronized(TasksLocalDataSource::javaClass) {
                    INSTANCE = TasksLocalDataSource(context)
                }
            }
            return INSTANCE!!
        }
    }
```